modules/placement-driver/src/test/java/org/apache/ignite/internal/placementdriver/LeaseUpdaterTest.java

```diff
@@ -301,11 +288,69 @@ public void testLeaseAmongPendings() throws Exception {
         Lease lease = awaitForLease();
 
         assertEquals(pendingNode.name(), lease.getLeaseholder());
+    }
 
-        leaseUpdater.deactivate();
+    @Test
+    public void testLeaseAmongPendingsSkipsLearners() throws Exception {
+        var peer = Assignment.forPeer("test-node-pending-peer");
+        var learner = Assignment.forLearner("test-node-pending-learner");
+
+        // no stable, pending peer and learner
+        mockTopology(Set.of(), Set.of(peer, learner));
+
+        Lease lease = awaitForLease();
+
+        assertEquals(peer.consistentId(), lease.getLeaseholder());
+
+        // no stable, pending learner only
+        mockTopology(Set.of(), Set.of(learner));
+
+        assertThrows(AssertionError.class, this::awaitForLease);
```

Comment:
This check will last 10 seconds, which is not needed. Maybe add an overloaded method with timeout and call it here with 1 sec value?